### DISCUSSION TITLE
Contributing to Open MCT

### DISCUSSION THREAD

**Discussion by jvigliotta** [nasa/openmct#3821] [Announcements]
If you would like to contribute code to Open MCT, please take a look at [our contribution guidelines](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md), there you'll find information on standards for commits/PRs as well as style guides and best practices. Also, all contributors must have a signed [Contributor License Agreement (CLA)](https://nasa.github.io/openmct/static/files/ind-cla-open-mct.pdf) in our records before we can approve/merge in your code. Any questions, please feel free to ask in [our Community Q&A](https://github.com/nasa/openmct/discussions/categories/community-q-a).

Once you've been notified your CLA is on file, feel free to take a look at our [help wanted](https://github.com/nasa/openmct/issues?q=is%3Aissue+is%3Aopen+label%3A%22help_wanted%22) issues that are currently not being worked on.

**Notes**
_- completed CLAs can be sent to email at the bottom of the 2nd page of the CLA_
_- it usually takes around a week (in some cases, up to a month or two) turnaround time for processing_
_- be sure to hand sign or electronically sign the CLA_
_- instructions on "Fill & Sign" [here](https://www.adobe.com/acrobat/how-to/fill-sign-pdf-forms-electronically.html)_

**Comment by JonanOribe** [comment]
Hi @jvigliotta, where should I send the signed CLA??

**Reply by kamilkurach** [reply]
Hi @JonanOribe I guess `Kim.L.Chrestenson@nasa.gov` is a contact person. Check the bottom of 2 page of CLA.

**Reply by JonanOribe** [reply]
Thanks, my fault, the information is right there

**Reply by jvigliotta** [reply]
Just a note, for anyone reading up on this, **`hong.v.vong@nasa.gov`**, is the new email to send the CLA's to.

**Comment by jvigliotta** [comment]
Hi @JonanOribe , no worries, @yacotaco is correct, you can send those **`hong.v.vong@nasa.gov`**. Once he receives them, he'll notify us and we'll add you to our list of approved contributors.

_Note: Email updated 3/31/26 to reflect staffing changes._

**Reply by JonanOribe** [reply]
Hi @jvigliotta  I sent the CLA weeks ago, but until now I have no response. How can I know if everything its ok??. Thank you

**Reply by jvigliotta** [reply]
@JonanOribe I'll check in with them on the status.

**Reply by jvigliotta** [reply]
Ok, checked in and they're not seeing it. Can you resend it to the original email and CC `hong.v.vong@nasa.gov` ?

**Reply by aimanafzal** [reply]
Hi @jvigliotta, I have sent a signed CLA yesterday to `Kim.L.Chrestenson@nasa.gov`. 
Usually, in how many days I should expect to hear back from them?

**Reply by jvigliotta** [reply]
Hi @aimanafzal! It usually takes around a week, but sometimes longer. Feel free to check-in with me if it's been longer than a week. Thanks!

**Reply by aimanafzal** [reply]
Hi @jvigliotta , Its been more than a month since I have sent a signed copy of NDA and still not heard back. 
Does that mean, I can start contributing?

Looking forward.

Thanks,
Aiman

**Reply by jvigliotta** [reply]
hi @aimanafzal I'm waiting for an updated list of CLA's I will let you know when I hear back!

**Reply by aimanafzal** [reply]
Hi @jvigliotta, any updates so far?

Thanks

**Comment by iSagnik** [comment]
@jvigliotta Hi, I am developing a project with OpenMCT as well that involves a timeline and was having trouble adding some plugins/custom features. I would appreciate it if I could have a one on one discussion with you to ask some questions if you can make the time! This is project os for a student rocketry team called Liquid Rocketry Lab ([Liquid Rocketry Lab](liquidrocketry.com))

**Comment by gholin** [comment]
@jvigliotta I was trying to sign the CLA digitally, but once I downloaded it and tried to add the signature, it said I could not add the signature because it was blocked from modification by the owner. Oddly enough, I was able to modify the other fields and type my address and name in. Any help would be great! Thanks!

**Reply by jvigliotta** [reply]
I think the issue may be that you need to "Fill & Sign" the document: https://www.adobe.com/acrobat/how-to/fill-sign-pdf-forms-electronically.html if you try to edit the document directly, it will show the ownership error message.

**Comment by LuciNyan** [comment]
Before I signed, he prompted me to fill in the owner's password first. What is the owner's password?  🤔️

**Reply by jvigliotta** [reply]
Hi @LuciNyan ! Sorry for the confusion. We're going to look into what's causing that. A workaround in the meantime, is to print it out and sign it, then scan it in and send that off. Once we have an update regarding the owner password issue, we'll update on this thread. Thank you!

**Reply by jvigliotta** [reply]
I think the issue may be that you need to "Fill & Sign" the document: https://www.adobe.com/acrobat/how-to/fill-sign-pdf-forms-electronically.html if you try to edit the document directly, it will show the ownership error message.

**Reply by LuciNyan** [reply]
Hi! @jvigliotta. Thanks a lot. It seems that the problem lies in the default PDF reader of mac. When I changed to Adobe Reader, the problem was solved

**Reply by LuciNyan** [reply]
Hi @jvigliotta! 
I've been sending the CLA email for over a week and haven't heard back. I wonder if there is anything else I need to do.

**Comment by ewferg** [comment]
[jvigliotta](https://github.com/jvigliotta), do you have a governance structure for how decisions are made regarding the direction of the project amongst key community members and stakeholders?

**Reply by akhenry** [reply]
This is a really great question. Currently the project is under the stewardship of the Open MCT team at NASA. The project direction is largely dictated by NASA mission needs because we are largely funded by individual missions. We encourage input from the community both in the form of code contributions and direction.

**Comment by tristan-f-r** [comment]
By the way, is the choice not to use a CLA automation tool such as [CLA assistant](https://cla-assistant.io/) deliberate? I assume it has to do with legal restrictions?

**Reply by ozyx** [reply]
@LeoDog896 That's correct, right now the process is very much manually operated. We have discussed exploring automated options but it will depend on what we're allowed to do legally.

**Comment by saram7619** [comment]
Hi, I emailed my CLA last Monday and I was wondering if it has been approved.

**Reply by ozyx** [reply]
It has been approved! Thanks for your patience.

**Comment by NickMezacapa** [comment]
Hi, I just found another workaround for those experiencing the 'ownership password' issue when annotating the CLA. (in macOS Preview)

I solved it by exporting the file with a new owner password added. Here are the steps:

- Save the file to your computer, then open with Preview
- File -> Export
- Click on 'Permissions' button
- Add/verify a new password at the bottom of the modal, do not check anything else. Then click 'Apply'
- Save the file after adding ownership, and replace current file if you do not change the file name
- Reopen the document and try to add your signature annotation again
- Enter the password you just created, should work now

**Reply by ozyx** [reply]
Awesome! Thanks for sharing!

**Comment by Harshikerfuffle** [comment]
Kinda new to open source, but this project looks really cool. Before signing up to be a contributor, where can I see a list of areas where contributions are needed? I couldn't access the 'projects' page

**Reply by ozyx** [reply]
@Harshikerfuffle Have a look through our GitHub issues! A good place to start would be with [any issues marked `help wanted`](https://github.com/nasa/openmct/issues?q=is%3Aopen+is%3Aissue+label%3Ahelp_wanted). Also check out our [Open e2e Testing Initiative](https://github.com/nasa/openmct/discussions/4303). There are plenty of test stubs in the codebase that just need to be written out (search the codebase for `test.fixme` to find the stubs). 

I do recommend sending in the CLA form ASAP as the process of approval is manual and can take a variable amount of time (sometimes a month or longer). We just need an approved CLA on file in order to merge the code. You can still go ahead and start (and we can help / review code) while you wait for that to clear.

Thanks for your interest and let us know if you have any questions or need anything while getting started!

**Reply by Harshikerfuffle** [reply]
thanks a bunch for sharing. I will start working on the CLA :)

**Comment by scottrobel** [comment]
Hello friends!
I submitted the Individual Contributor Agreement just over a month ago, and I haven't heard anything back.
Would anyone be able to check if my agreement is on file yet? If not, would it be okay to start working on issues while it's pending?
Cheers,
Scott

**Reply by ozyx** [reply]
Hi @scottrobel,

Thanks for your interest in contributing to Open MCT! Yes, you are OK to start work on open issues and open PRs. We can even review your PRs before the CLA has been approved. I'll go ahead and follow up with my contacts on the status of your CLA in the meantime.

Let us know if you have any questions or if there's anything we can do to help you get started.

Cheers,
Jesse

**Reply by ozyx** [reply]
@scottrobel Your CLA has been processed and approved! You are good to go 🚀

**Comment by kanishk6103** [comment]
Hey! I submitted the Individual Contributor Agreement a few weeks ago and I have not received any reply. I have raised a Pull request, will that be merged only when my CLA is approved? @ozyx 
My PR #8201 

Regards,
Kanishk

**Comment by vijaygovindaraja** [comment]
Hi, I've submitted my signed Individual CLA via email to the Ames Software Release Team. My GitHub username is vijaygovindaraja.

I currently have the following pull requests open:

- #8293 — fix: defer Time Conductor mode change until confirm button is clicked
- #8294 — Add DSN telemetry plugin for real-time Deep Space Network data
- #8295 — Add orbital tracking plugin with real-time satellite visualization
- #8296 — feat: migrate core time formatters from Moment.js to Luxon (Part 1)

Looking forward to contributing more. Thanks!

**Comment by LeonardoSantosBR** [comment]
Hey! I submitted the Individual Contributor Agreement and I saw some comments saying that the response is taking a long time; in the meantime, is it possible to contribute to the project? @jvigliotta

**Comment by danfry1** [comment]
Hi — I emailed my signed Individual CLA to the Ames Software Release Team today (20 August 2026). My GitHub username is **danfry1**.

I'm planning to contribute to the time-handling work (#7872 / #7393). Could you confirm once the agreement is on file? Thanks!

**Comment by dorofeeva-maria** [comment]
Hi! Quick question — is @ozyx  2024 advice [here](https://github.com/nasa/openmct/discussions/3821#discussioncomment-7994011) (help wanted issues, the e2e Initiative and test.fixme stubs) still the best way to start in 2026?

Most small bugs already have open PRs and the e2e thread's been quiet since 2023, so I wasn't sure if those stubs are still wanted or just old. I know Vue 3 and Playwright and want to make a solid first PR — is there a current area where help would be most useful? Happy to do Vue, accessibility, or e2e. Thanks!
